modules/core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/event/impl/EventServiceImpl.java

```diff
@@ -117,33 +124,35 @@ void push(Event event) {
 
     protected boolean bulkModeEnabled = false;
 
-    protected EventPipeRegistry registeredPipes = new EventPipeRegistry();
-
-    protected EventDispatcherRegistry dispatchers = new EventDispatcherRegistry();
-
     protected EventBundleDispatcher pipeDispatcher;
 
-    // @since 11.4
-    protected DescriptorRegistry domainEventProducers = new DescriptorRegistry();
-
-    // @since 11.4
-    protected static final String REGISTRY_TARGET_NAME = "EventService";
-
     public EventServiceImpl() {
-        listenerDescriptors = new EventListenerList();
         postCommitExec = new PostCommitEventExecutor();
         asyncExec = new AsyncEventExecutor();
     }
 
+    @SuppressWarnings("unchecked")
+    protected <T extends Registry> T getRegistry(String extensionPoint) {
+        return (T) Framework.getRuntime()
+                            .getComponentManager()
+                            .getExtensionPointRegistry(NAME, extensionPoint)
+                            .orElseThrow(() -> new IllegalArgumentException(String.format(
+                                    "Unknown registry for extension point '%s--%s'", NAME, extensionPoint)));
+
+    }
+
     public void init() {
         asyncExec.init();
 
-        EventDispatcherDescriptor dispatcherDescriptor = dispatchers.getDispatcherDescriptor();
-        if (dispatcherDescriptor != null) {
-            List<EventPipeDescriptor> pipes = registeredPipes.getPipes();
+        listenerDescriptors = this.<EventListenerList> getRegistry(EVENT_LISTENER_XP);
+        Optional<EventDispatcherDescriptor> optDispatcher = this.<SingleRegistry> getRegistry(EVENT_DISPATCHER_XP)
+                                                                .getContribution();
+        if (optDispatcher.isPresent()) {
```

Comment:
If there's no checked exception messing things up, this `if` block would be a lambda of `ifPresent` on the previous call.